util/generate-rsa-signing-tests.py

```diff
@@ -40,57 +56,163 @@ def parse(fn):
                 continue
             name, val = ln.split('=', 1)
             cur[name.strip()] = val.strip()
-            if name.strip() == 'S':
+            if name.strip() == last_field:
                 cases.append(cur)
                 cur = copy.copy(cur)
     return cases
 
-def main(fn):
-    for case in parse(fn):
+def print_sign_test(case, n, e, d, padding_alg):
+    # Can only support moduli which are a whole number of bytes
+    if n.bit_length() % 8 != 0:
+        debug("Skipping due to modulus (not a whole number of bytes).", DEBUG)
+        return
```

Comment:
Sorry, I gave you bad advice. In the review of #283 we did the math and found out that even if we require the private key primes `p` and `q` to be multiples of 8 bits or even multiples of 1024 bits, it's still possible and even pretty likely that the public modulus will be 2047 bits; i.e. `n.bit_length() % 8 == 7`. So, we in fact do need to support public moduli that aren't an even number of bytes, even for signing. Sorry!